src/function/scalar/string/regexp.cpp

```diff
@@ -203,6 +203,104 @@ static unique_ptr<FunctionData> RegexReplaceBind(ClientContext &context, ScalarF
+}
+
+static void RegexExtractFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	auto &func_expr = (BoundFunctionExpression &)state.expr;
+	const auto &info = (RegexpExtractBindData &)*func_expr.bind_info;
+
+	auto &strings = args.data[0];
+	auto &patterns = args.data[1];
+	if (info.constant_pattern) {
+		UnaryExecutor::Execute<string_t, string_t>(strings, result, args.size(), [&](string_t input) {
+			return Extract(input, result, info.re, info.rewrite);
+		});
+		//		if (info.constant_group) {
+		//
+		//		} else {
+		//			BinaryExecutor::Execute<string_t, int32_t, string_t>(
+		//			    strings, groups, result, args.size(), [&](string_t input, int32_t group) {
+		//				    string group_string = "\\" + std::to_string(MinValue(group, 9));
+		//				    auto rewrite = duckdb_re2::StringPiece(group_string);
+		//				    return Extract(input, result, info.re, rewrite);
+		//			    });
+		//		}
+	} else {
+		BinaryExecutor::Execute<string_t, string_t, string_t>(strings, patterns, result, args.size(),
+		                                                      [&](string_t input, string_t pattern) {
+			                                                      RE2 re(CreateStringPiece(pattern));
+			                                                      return Extract(input, result, re, info.rewrite);
+		                                                      });
+		//		if (info.constant_group) {
+		//
+		//		} else {
+		//			TernaryExecutor::Execute<string_t, string_t, int32_t, string_t>(
+		//			    strings, patterns, groups, result, args.size(), [&](string_t input, string_t pattern, int32_t
+		// group) { 				    RE2 re(CreateStringPiece(pattern)); 				    string group_string =
+		// "\\"
+		// + std::to_string(group); 				    duckdb_re2::StringPiece rewrite(group_string); return
+		// Extract(input, result, re, rewrite);
+		//			    });
+		//		}
+	}
+}
+
+RegexpExtractBindData::RegexpExtractBindData(bool constant_pattern, const string &pattern, const string &group_string_p)
+    : constant_pattern(constant_pattern), re(duckdb_re2::StringPiece(pattern)), group_string(group_string_p),
+      rewrite(group_string) {
+}
+
+unique_ptr<FunctionData> RegexpExtractBindData::Copy() {
+	return make_unique<RegexpExtractBindData>(constant_pattern, re.pattern(), group_string);
+}
+
+static unique_ptr<FunctionData> RegexExtractBind(ClientContext &context, ScalarFunction &bound_function,
+                                                 vector<unique_ptr<Expression>> &arguments) {
+	D_ASSERT(arguments.size() >= 2);
+
+	bool constant_pattern = arguments[1]->IsFoldable();
+	string pattern = "";
+	if (constant_pattern) {
+		Value pattern_str = ExpressionExecutor::EvaluateScalar(*arguments[1]);
+		if (!pattern_str.is_null && pattern_str.type().id() == LogicalTypeId::VARCHAR) {
+			pattern = pattern_str.str_value;
+		}
+	}
+
+	string group_string = "";
+	if (arguments.size() == 3) {
+		bound_function.arguments[2] = LogicalType::INTEGER;
```

Comment:
This should not be required, and the comment below can be removed as well right?